packages/server/integrations/updatePrevUsedRepoIntegrationsCache.ts

```diff
@@ -0,0 +1,31 @@
+import IntegrationRepoId from 'parabol-client/shared/gqlIds/IntegrationRepoId'
+import getAllCachedRepoIntegrations from '../graphql/queries/helpers/getAllCachedRepoIntegrations'
+import getPrevUsedRepoIntegrations from '../graphql/queries/helpers/getPrevUsedRepoIntegrations'
+import getPrevUsedRepoIntegrationsRedisKey from '../utils/getPrevUsedRepoIntegrationsRedisKey'
+import getRedis from '../utils/getRedis'
+
+const updatePrevUsedRepoIntegrationsCache = async (teamId: string, repoIntegrationId: string) => {
+  const redis = getRedis()
+  const prevUsedRepoIntegrationsKey = getPrevUsedRepoIntegrationsRedisKey(teamId)
+  const [prevUsedRepoIntegrations, allCachedRepoIntegrations] = await Promise.all([
+    getPrevUsedRepoIntegrations(teamId),
+    getAllCachedRepoIntegrations(teamId)
```

Comment:
What scenario are you thinking of where that might happen?

When `repoIntegrations` is initially called, it validates the cache. `updatePrevUsedRepoIntegrationsCache` is called when the user has selected a `repoIntegration` from the menu.

It is possible that the user: 
- opens the menu
- `allRepoIntegrationsCache` is updated
- someone then deleted a repo in GitHub, for example
- the user selects the deleted repo from the cache, but it doesn't exist

However, in that case (I just tested it), we'd get an error message on the task saying that the repo doesn't exist. This function wouldn't be called so it wouldn't be added to the prevUsed cache, and when `repoIntegrations` next fetches from the network, it would remove that repo.